### DISCUSSION TITLE
Specify a minimum zoom level

### DISCUSSION THREAD

**Discussion by carderne** [developmentseed/titiler#512] [General]
(Preface: I think it's quite likely that I've misunderstood something!)

We're using titiler (via titiler-pgstac) with source data at 10 m/pixel (z~=14) with four overviews down to z~=10.
Tiles at lower zoom levels are very slow, and tiles below z=5 or so would probably take minutes or hours to generate - jamming up the machine for no reason.

As far as I can tell, there's no built-in way to ignore requests below a certain zoom level? (Something analogous to `tilejson`'s `minzoom`.)

Currently I'm wrapping the functions with something like the code below. Happy to work on a PR if this would be useful for anyone, otherwise keen to just understand titiler better!
```python
if kwargs["z"] <= 9:
    return Response(status_code=status.HTTP_204_NO_CONTENT)
```

**Comment by vincentsarago** [comment]
👋 @carderne, in titiler-pgstac you can add a `minzoom` information in the mosaic metadata (when you register your mosaic, https://stac-utils.github.io/titiler-pgstac/advanced/metadata/) which will then be used as minzoom in the tilejson document https://github.com/stac-utils/titiler-pgstac/blob/master/titiler/pgstac/factory.py#L280

Sadly for STAC we can't know what's the minzoom for the datasource in advance. 

The mosaic metadata is only retrieved in the tilejson endpoint because we wanted to avoid unnecessary database request on each tile request

**Comment by carderne** [comment]
Thanks @vincentsarago 😊! I think I’d mostly figured that out, good to understand the reasoning.

Our users include ArcGIS users, who I don’t think can consume tilejson (more research needed here) so need a plain XYZ endpoint.

So the problem with my solution is clearly that the zoom level is hardcoded into the paths. I might add some end vars or something if we need a different `z` cutoff per `searchid` but it’s definitely not very ergonomic or extensible…

**Reply by vincentsarago** [reply]
I'll try to write something in titiler-pgstac over the weekend to show how you can fetch the metadata on each tile request and then return empty if needed

**Reply by carderne** [reply]
Sorry, just realised my GitHub Discussion faux pas of starting another thread instead of writing a reply...

Are you talking about fetching metadata from the db? I'm pretty sure we'd also want to avoid that, but could be interesting to explore.

Thanks for the amazing libraries btw!
